### DISCUSSION TITLE
Detecting undef on HTML

### DISCUSSION THREAD

**Discussion by nneto** [vuejs/language-tools#1789] [Q&A] [answered]
I'm no sure where to ask this question (if here or eslint repo, or somewhere else). But I was wondering if is it possible to detect undefined variables and functions in the HTML (in this case exemplified below, highlight `bar`), in the same way the `foo` is detected and underlined in the script part.

![foo](https://user-images.githubusercontent.com/7104467/187322178-efefb90e-7b82-4540-99a8-401fe78da0f0.png)

I just don't know if I'm missing a setting, or if it's a missing feature or even if it's just something that can't be done for not being possible to infer this kind of thing with linting.

**Comment by johnsoncodehk** [comment, marked as answer]
Since you didn't use `lang="ts"`, you need to enable `checkJs` in tsconfig.json / jsconfig.json, or just put `// @ts-check` in script block top.
